packages/x-date-pickers/src/PickersLayout/PickersLayout.tsx

```diff
@@ -23,24 +23,68 @@ const PickersLayoutRoot = styled('div', {
   name: 'MuiPickersLayout',
   slot: 'Root',
   overridesResolver: (props, styles) => styles.root,
-})<{ ownerState: { isLandscape: boolean } }>(({ theme, ownerState }) => ({
+})<{ ownerState: { isLandscape: boolean; isRTL: boolean } }>({
   display: 'grid',
   gridAutoColumns: 'max-content auto max-content',
   gridAutoRows: 'max-content auto max-content',
-  [`& .${pickersLayoutClasses.toolbar}`]: ownerState.isLandscape
-    ? {
-        gridColumn: theme.direction === 'rtl' ? 3 : 1,
-        gridRow: '2 / 3',
-      }
-    : { gridColumn: '2 / 4', gridRow: 1 },
-  [`.${pickersLayoutClasses.shortcuts}`]: ownerState.isLandscape
-    ? { gridColumn: '2 / 4', gridRow: 1 }
-    : {
-        gridColumn: theme.direction === 'rtl' ? 3 : 1,
-        gridRow: '2 / 3',
-      },
   [`& .${pickersLayoutClasses.actionBar}`]: { gridColumn: '1 / 4', gridRow: 3 },
-}));
+  variants: [
+    {
+      props: { isLandscape: true },
+      style: {
+        [`.${pickersLayoutClasses.shortcuts}`]: {
+          gridColumn: '2 / 4',
+          gridRow: 1,
+        },
+      },
+    },
+    {
+      props: { isLandscape: false },
+      style: {
+        [`& .${pickersLayoutClasses.toolbar}`]: {
+          gridColumn: '2 / 4',
+          gridRow: 1,
+        },
+      },
+    },
+    {
+      props: { isLandscape: true, isRTL: false },
```

Comment:
We have a position that depends on both the `isLandscape` prop and the `isRTL` (or `theme.direction`, see the related comment for context) prop.
Is there a way to avoid creating a variant for each combination? My first approach was to create a CSS variable and change its value in a variant depending on `isRTL` and then only have variants that depends on `isLandscape` that uses this CSS variable. But this does not seem to work....
I also tried nested variants but same, it's not supported (and for good reasons I guess :laughing: )

Reply:
I did not thought about it, thanks!